ilp-core/src/test/java/org/interledger/core/InterledgerAddressTest.java

```diff
@@ -196,6 +196,35 @@ public void testStartsWithInterledgerAddress() {
     assertThat(address.startsWith(InterledgerAddress.of("test1.foo.bob")), is(false));
   }
 
+  @Test
+  public void testStartsWithInterledgerAddressPrefix() {
+    final InterledgerAddress address = InterledgerAddress.of("g.foo.bob");
+    assertThat(address.startsWith(InterledgerAddressPrefix.of("g")), is(true));
+    assertThat(address.startsWith(InterledgerAddressPrefix.of("g.foo")), is(true));
+    assertThat(address.startsWith(InterledgerAddressPrefix.of("g.foo.bob")), is(true));
+    assertThat(address.startsWith(InterledgerAddressPrefix.of("g.foo.bar")), is(false));
+    assertThat(address.startsWith(InterledgerAddressPrefix.of("test.foo")), is(false));
+
+    final InterledgerAddress smallAddress = InterledgerAddress.of("g.foo");
+    assertThat(smallAddress.startsWith(InterledgerAddressPrefix.of("g")), is(true));
+    assertThat(smallAddress.startsWith(InterledgerAddressPrefix.of("g.foo")), is(true));
+    assertThat(smallAddress.startsWith(InterledgerAddressPrefix.of("g.foo.bob")), is(false));
+    assertThat(smallAddress.startsWith(InterledgerAddressPrefix.of("g.foo.bar")), is(false));
+    assertThat(smallAddress.startsWith(InterledgerAddressPrefix.of("test.foo")), is(false));
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testStartsWithInterledgerAddressPrefixWithNull() {
```

Comment:
I think this test is slightly incorrect. It's actually failing with an NPE inside of `InterledgerAddressPrefix`, which is not what we want to be testing here. 

Instead, this should probably be:

```java
 @Test(expected = NullPointerException.class)
  public void testAddressStartsWithNull() {
    final InterledgerAddress address = InterledgerAddress.of("g.foo");
    InterledgerAddressPrefix addressPrefix = null;
    try {
      address.startsWith(addressPrefix); // addressPrefix will be null
    } catch (NullPointerException e) {
      assertThat(e.getMessage(), is("addressPrefix must not be null!"));
      throw e;
    }
  }
```